go/vt/vtgate/planbuilder/operators/subquery_planning.go

```diff
@@ -42,11 +42,11 @@ func isMergeable(ctx *plancontext.PlanningContext, query sqlparser.SelectStateme
 
 	switch node := query.(type) {
 	case *sqlparser.Select:
-		if len(node.GroupBy) > 0 {
+		if node.GroupBy != nil && len(node.GroupBy.Exprs) > 0 {
```

Comment:
Since we are setting group by to `nil` everytime we empty it, it should be fine to just check for `node.GroupBy != nil`.